plugins/modules/system/keyring.py

```diff
@@ -0,0 +1,282 @@
+    This is primarily used for 'headless' hosts.
+    """
+    try:
+        keyring.delete_password(module.params["service"], module.params["username"])
+        return True
+    except keyring.errors.KeyringLocked as keyring_locked_err:  # pylint: disable=unused-variable
+        delete_argument = (
+            'echo "%s" | gnome-keyring-daemon --unlock\nkeyring del %s %s\n'
+            % (
+                quote(module.params["keyring_password"]),
+                quote(module.params["service"]),
+                quote(module.params["username"]),
+            )
+        )
+        alternate_delete = module.run_command(
+            "dbus-run-session -- /bin/bash",
+            use_unsafe_shell=True,
+            data=delete_argument,
+            encoding=None,
+        )
+
+        if alternate_delete[2].decode("UTF-8"):
+            return alternate_delete[2].decode("UTF-8")
+        if not alternate_delete[2].decode("UTF-8"):
+            return True
+
+
+def set_passphrase(module):
+    """
+    Attempt to set passphrase in the keyring using a shell rather than the Python API.
+    This is primarily used for 'headless' hosts.
+    """
+    try:
+        keyring.set_password(
+            module.params["service"],
+            module.params["username"],
+            module.params["user_password"],
+        )
+        return True
+    except keyring.errors.KeyringLocked as keyring_locked_err:  # pylint: disable=unused-variable
+        set_argument = (
+            'echo "%s" | gnome-keyring-daemon --unlock\nkeyring set %s %s\n%s\n'
+            % (
+                quote(module.params["keyring_password"]),
+                quote(module.params["service"]),
+                quote(module.params["username"]),
+                quote(module.params["user_password"]),
+            )
+        )
+        alternate_set = module.run_command(
+            "dbus-run-session -- /bin/bash",
+            use_unsafe_shell=True,
+            data=set_argument,
+            encoding=None,
+        )
+
+        try:
+            return alternate_set[1].decode("UTF-8").splitlines()
+        except IndexError:
+            return True
+
+
+def get_passphrase(module):
+    """
+    Attempt to retrieve passphrase from keyring using a shell rather than the Python API.
+    This is primarily used for 'headless' hosts.
+    """
+    try:
+        passphrase = keyring.get_password(
+            module.params["service"], module.params["username"]
+        )
+        return passphrase
+    except keyring.errors.KeyringLocked as keyring_locked_err:  # pylint: disable=unused-variable
+        pass
+    except AttributeError as attr_err:  # pylint: disable=unused-variable
+        pass
+    get_argument = 'echo "%s" | gnome-keyring-daemon --unlock\nkeyring get %s %s\n' % (
+        quote(module.params["keyring_password"]),
+        quote(module.params["service"]),
+        quote(module.params["username"]),
+    )
+    get_result = module.run_command(
```

Comment:
```suggestion
    dummy, stderr, dummy = module.run_command(
```

Reply:
Have updated this to make use of the `stderr`.
`stderr` is the third element.

>A 3-tuple of return code (integer), stdout (native string), and stderr (native string)

https://docs.ansible.com/ansible/latest/reference_appendices/module_utils.html

Reply:
Ah sorry, in that case some of my other comments need to be adjusted as well...